packages/@react-aria/tooltip/test/useTooltip.test.js

```diff
@@ -10,18 +10,211 @@
  * governing permissions and limitations under the License.
  */
 
+import {act, fireEvent, render} from '@testing-library/react';
 import React from 'react';
-import {renderHook} from '@testing-library/react-hooks';
-import {useTooltip} from '../';
-
-describe.skip('useTooltip', function () {
-  let renderTooltipHook = (props) => {
-    let {result} = renderHook(() => useTooltip(props));
-    return result.current;
-  };
-
-  it('handles defaults', function () {
-    let {tooltipProps} = renderTooltipHook({children: 'Test Tooltip'});
-    expect(tooltipProps.role).toBe('tooltip');
+import {useInteractionModality} from '@react-aria/interactions';
+import {useTooltip, useTooltipTrigger} from '../';
+import {useTooltipTriggerState} from '@react-stately/tooltip';
+
+describe('useTooltip', function () {
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+  it('opens tooltip on hover after delay', function () {
+    jest.useFakeTimers('modern');
+
+    function Test() {
+      useInteractionModality();
+
+      const state = useTooltipTriggerState();
+      const {triggerProps, tooltipProps} = useTooltipTrigger({}, state);
+      const {tooltipProps: finalTooltipProps} = useTooltip(tooltipProps);
+
+      return (
+        <div>
+          <span {...triggerProps}>{'Trigger'}</span>
+          {state.isOpen && <span {...finalTooltipProps}>{'Tooltip'}</span>}
+        </div>
+      );
+    }
+
+    const {container, queryByRole} = render(<Test />);
+    const triggerElement = container.firstChild.firstChild;
+
+    // trigger pointer modality
+    fireEvent.mouseMove(container);
+
+    fireEvent.mouseEnter(triggerElement);
+
+    expect(queryByRole('tooltip')).toBeNull();
+    act(jest.runAllTimers);
+    expect(queryByRole('tooltip')).not.toBeNull();
+  });
+  it('opens tooltip immediately on hover with `delay: 0`', function () {
+    function Test() {
+      useInteractionModality();
+
+      const state = useTooltipTriggerState({delay: 0});
+      const {triggerProps, tooltipProps} = useTooltipTrigger({}, state);
+      const {tooltipProps: finalTooltipProps} = useTooltip(tooltipProps);
+
+      return (
+        <div>
+          <span {...triggerProps}>{'Trigger'}</span>
+          {state.isOpen && <span {...finalTooltipProps}>{'Tooltip'}</span>}
+        </div>
+      );
+    }
+
+    const {container, queryByRole} = render(<Test />);
+    const triggerElement = container.firstChild.firstChild;
+
+    // trigger pointer modality
+    fireEvent.mouseMove(container);
+
+    fireEvent.mouseEnter(triggerElement);
+
+    expect(queryByRole('tooltip')).not.toBeNull();
+  });
+
+  it('hides tooltip when hover leaves the trigger', function () {
+    jest.useFakeTimers('modern');
+
+    function Test() {
+      useInteractionModality();
+
+      const state = useTooltipTriggerState({delay: 0});
+      const {triggerProps, tooltipProps} = useTooltipTrigger({}, state);
+      const {tooltipProps: finalTooltipProps} = useTooltip(tooltipProps);
```

Comment:
we should pass `state` to `useTooltip` given your comment about this should be required? same comment for the other instances in this file
or is this to make sure that it works both ways?

Reply:
> or is this to make sure that it works both ways?

This. As mentioned in some other comment - I think that this should be made required in the future but that would be a breaking change. We could add a single test ensuring that though and just make the rest use `state` - I'm fine either way, just let me know what do you think is best here.

Reply:
I think we can leave as is then, I just wanted to make sure it was intentional.